backend/src/packages/users/user.repository.ts

```diff
@@ -20,7 +26,8 @@ class UserRepository implements IRepository {
   }
 
   public async create(entity: UserEntity): Promise<UserEntity> {
-    const { email, passwordSalt, passwordHash } = entity.toNewObject();
+    const { email, passwordSalt, passwordHash, firstName, lastName } =
+      entity.toNewObject();
 
     const item = await this.userModel
       .query()
```

Comment:
You may insert userDetails joined to user like this
And declare this const value for user relation expression
```
private defaultRelationExpression = 'userDetails';
```

```suggestion
    const user = await this.userModel
      .query
      .insertGraphAndFetch({
        email,
        passwordSalt,
        passwordHash,
        userDetails: {
          firstName,
          lastName,
        },
      })
      .withGraphFetched(this.defaultRelationExpression)
      .execute();
```